src/mbgl/renderer/layers/symbol_layer_tweaker.cpp

```diff
@@ -131,7 +131,7 @@ void SymbolLayerTweaker::execute(LayerGroupBase& layerGroup,
         const auto& symbolData = static_cast<gfx::SymbolDrawableData&>(*drawable.getData());
         const auto isText = (symbolData.symbolType == SymbolType::Text);
 
-        if (isText && (textPaintBuffer || textPropertiesUpdated)) {
+        if (isText && (!textPaintBuffer || textPropertiesUpdated)) {
```

Comment:
Note: This was causing a UBO to be updated every time, and should improve performance.